utils/irqbalance/Makefile

```diff
@@ -51,7 +61,7 @@ CONFIGURE_ARGS+= \
 
 CONFIGURE_VARS += \
 	GLIB2_LIBS="$(STAGING_DIR)/usr/lib/libglib-2.0.a \
-		$(STAGING_DIR)/usr/lib/libiconv-stub/lib/libiconv.a"
+		$(TARGET_LDFLAGS)"
```

Comment:
Use ICONV_LDFLAGS

Reply:
In which case it would need to use both ICONV_LDFLAGS and INTL_LDFLAGS

Reply:
Sure. But TARGET_LDFLAGS is massive.

Why would INTL be needed?

Reply:
It needs it if BUILD_NLS is enabled in the config. If it's not linked then it errors out thus:

```
OpenWrt-libtool: link: ccache_cc -I/opt/openwrt/staging_dir/target-x86_64_glibc/usr/include/glib-2.0 -I/opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib/glib-2.0/include -O2 -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -I/opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib/libiconv-full/include -I/opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib/libintl-full/include -ffile-prefix-map=/opt/openwrt/build_dir/target-x86_64_glibc/irqbalance-1.6.0=irqbalance-1.6.0 -Wformat -Werror=format-security -D_FORTIFY_SOURCE=1 -Wl,-z -Wl,now -Wl,-z -Wl,relro -Wl,-rpath-link=/opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib/libiconv-full/lib -Wl,-rpath-link=/opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib/libintl-full/lib -znow -zrelro -o irqbalance activate.o bitmap.o classify.o cputree.o irqbalance.o irqlist.o numa.o placement.o procinterrupts.o -Wl,-rpath-link=/opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib/libiconv-full/lib -Wl,-rpath-link=/opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib/libintl-full/lib -znow -zrelro  -L/opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib -L/opt/openwrt/staging_dir/target-x86_64_glibc/lib -L/opt/openwrt/staging_dir/toolchain-x86_64_gcc-9.3.0_glibc/usr/lib -L/opt/openwrt/staging_dir/toolchain-x86_64_gcc-9.3.0_glibc/lib -L/opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib/libiconv-full/lib -L/opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib/libintl-full/lib /opt/openwrt/staging_dir/target-x86_64_glibc/usr/lib/libglib-2.0.a -liconv -lpthread -lm
/opt/openwrt/staging_dir/toolchain-x86_64_gcc-9.3.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/9.3.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: /opt/openwrt/tmp/irqbalance.YYXFkr.ltrans0.ltrans.o: in function `glib_gettext':
<artificial>:(.text+0xa488): undefined reference to `libintl_dgettext'
/opt/openwrt/staging_dir/toolchain-x86_64_gcc-9.3.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/9.3.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: <artificial>:(.text+0xa4b0): undefined reference to `libintl_bindtextdomain'
/opt/openwrt/staging_dir/toolchain-x86_64_gcc-9.3.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/9.3.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: <artificial>:(.text+0xa4c4): undefined reference to `libintl_bind_textdomain_codeset'
/opt/openwrt/staging_dir/toolchain-x86_64_gcc-9.3.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/9.3.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: <artificial>:(.text+0xa508): undefined reference to `libintl_textdomain'
/opt/openwrt/staging_dir/toolchain-x86_64_gcc-9.3.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/9.3.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: <artificial>:(.text+0xa518): undefined reference to `libintl_gettext'
collect2: error: ld returned 1 exit status
```

Reply:
Ah that's going away in the next version. glib2 2.64 uses its own bundled copy.

Reply:
If using ICONV_LDFLAGS and INTL_LDFLAGS, the change is not quite as succint. It requires an intermediate variable since there's no TARGET_LDFLAGS to append to. 

```
CONFIGURE_ARGS+= \
    --disable-numa \
    --with-libcap_ng=no \
    --with-systemd=no \
    --without-irqbalance-ui \
    --enable-static=glib2

GLIB2_LIBS="$(STAGING_DIR)/usr/lib/libglib-2.0.a"

ifeq ($(CONFIG_BUILD_NLS),y)
    GLIB2_LIBS += $(INTL_LDFLAGS) -lintl
    GLIB2_LIBS += $(ICONV_LDFLAGS) -liconv
endif

ifeq ($(CONFIG_USE_GLIBC),y)
    GLIB2_LIBS += -lpthread
endif

CONFIGURE_VARS += \
    GLIB2_LIBS="$(GLIB2_LIBS)"
```

Reply:
The 2.64 update will make all of this irrelevant, assuming irqbalance uses pkgconfig to find which libraries to link to.

I will take a look at this when I can.

Reply:
It does, although there's clearly a problem with it. This is why the GLIB2_LIBS override was apparently added to the package in the first place in [commit bdcb87](https://github.com/openwrt/packages/commit/bdcb872a40e4567bc57c45299a0df5efad65207f), as it wasn't finding libiconv. 

```
configure:  GLIB2_LIBS  linker flags for GLIB2, overriding pkg-config
```

Reply:
Never mind then.

Reply:
Ok, I've updated and rebased the commit and fixed the signed-off-by